text/0000-security-advisories.md

```diff
@@ -0,0 +1,297 @@
+to attach an arbitrary amount of so-called security advisories to crates they
+own.
+
+Each advisory gets assigned an ID that is unique within the set of advisories
+for the affected crate.  Every advisory should have a unique URL, for example
+`https://crates.io/crates/<crate>/advisory/<id>`, where `<id>` is the
+advisory's ID. On that URL a human-readable representation of the advisory
+should be stored.
+
+Other pages on Crates.io should link to those advisories prominently where
+appropriate.
+
+## Cargo
+
+### `cargo advisory`
+
+A command called `advisory` will be added to Cargo. Here is an excerpt of its help
+page:
+
+```
+$ cargo advisory --help
+Generate and upload security advisories for the given or the current crate.
+
+Usage:
+    cargo advisory [options] -- [<crate>]
+    --filename PATH      The filename to use. Defaults to `./Advisory.toml`.
+                         If `-` is given, generated advisories are printed to
+                         stdout and advisories to upload are read from stdin.
+
+    --vers VERSION       Versions to release this advisory for. Can be
+                         specified multiple times. Only valid in conjunction
+                         with --generate.
+
+    --upload/--generate  Whether to upload or generate a advisory. The default
+                         is to generate. These options are mutually exclusive.
+    [...]
+```
+
+Like `yank` it takes a `--vers` option, with two differences:
+
+- if a version is not specified, `advisory` will default to all existing
+  versions.
+
+- Version ranges such as `<1.2.6, >1.0.0` can be specified. This is comparable
+  to the syntax used for specifying dependencies in the `Cargo.toml`, with the
+  exception that `x.y.z` is not equivalent to `^x.y.z`, but means the exact
+  version.
+
+Here's the workflow:
+
+1. The user invokes `cargo advisory` without the `--upload` option. Cargo will
+   generate a file under `filename`. Cargo should abort if the file already
+   exists. The content looks like this:
+
+   ```
+   [vulnerability]
+   package = "mypackage"
+   versions = ["1.2.0", "1.2.3", "1.2.4", "1.2.5"]
+
+   # It is strongly recommended to request a CVE, or alternatively a DWF, and
+   # reference the assigned number here.
+   # - CVE: https://iwantacve.org/
+   # - DWF: https://distributedweaknessfiling.org/
+   dwf = false
+   # dwf = "CVE-YYYY-XXXX"
+   # dwf = ["CVE-YYYY-XXXX", "CVE-ZZZZ-WWWW"]
+
+   # URL to a long-form description of this issue, e.g. a blogpost announcing
+   # the release or a changelog entry (optional)
+   url = false
+
+   # Enter a short-form description of the vulnerability here. Preferrably a
+   # single paragraph (required)
+   description = """
+   
+   """
+   ```
+
+2. The user invokes `cargo advisory --upload`. Cargo verifies the passed file
+   against the following rules:
+
+   - the file exists and is valid TOML
+   - When an optional key is `false`, this is semantically equivalent to it
+     being omitted.
```

Comment:
This makes no sense to me. If an advisory has no URL, then it just doesn't have an URL, so just leave it away.
But it certainly does **not** have a default URL with the value `false`.

This feels like some ugly JavaScript where you can use strings and booleans interchangeably, with arbitrary conversion rules.